### PR TITLE
backport: Merge bitcoin#24946, #22649, #22704, #23806, #26153

### PR DESCRIPTION
**Pull request by ogabrielides**
## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: test/sanitizer_suppressions/ubsan [hunk 1]
```diff
@@ -73,3 +74,9 @@ implicit-signed-integer-truncation:test/skiplist_tests.cpp
 implicit-signed-integer-truncation:torcontrol.cpp
 implicit-unsigned-integer-truncation:crypto/*
 implicit-unsigned-integer-truncation:leveldb/*
+shift-base:arith_uint256.cpp
+shift-base:crypto/
+shift-base:hash.cpp
+shift-base:streams.h
+shift-base:util/bip32.cpp
```
Comment (knst) [inline test/sanitizer_suppressions/ubsan:81]: lines with `arith_uint256...bip32` seems unrelated to backport.

**Review by knst — APPROVED** [review]
utACK

**Comment by UdjinM6** [comment]
26153: missing changes in `src/test/fuzz/crypto_diff_fuzz_chacha20.cpp` - should be partial.

File: test/sanitizer_suppressions/ubsan (hunk 1 above)
Comment (UdjinM6) [reply test/sanitizer_suppressions/ubsan:81]: @ogabrielides why?

**Comment by ogabrielides** [comment]
> 26153: missing changes in `src/test/fuzz/crypto_diff_fuzz_chacha20.cpp` - should be partial. Also needs rebase.

This file wasn't there. Unless if it was backported in the meanwhile (and that's why I need rebase)

**Comment by UdjinM6** [comment]
We don't have that file, so backport should be marked partial. And I asked for a rebase cause you'd have to change the name of the commit anyway and we need to fix `Merge Fast-Forward Only` check here.

**Comment by UdjinM6** [comment]
or you can only change commit name and I'll rebase it from GH GUI afterwards

**Comment by UdjinM6** [comment]
or even better - backport 22704 and 23806 (as discussed in Slack, should be trivial) and add missing changes to 26153

File: test/sanitizer_suppressions/ubsan [hunk 2]
```diff
@@ -29,6 +29,7 @@ unsigned-integer-overflow:stl_bvector.h
 unsigned-integer-overflow:txmempool.cpp
 unsigned-integer-overflow:util/strencodings.cpp
 unsigned-integer-overflow:validation.cpp
+unsigned-integer-overflow:xoroshiro128plusplus.h
```
Comment (knst) [inline test/sanitizer_suppressions/ubsan:32]: this time is in opposite missing one exception:

```
shift-base:xoroshiro128plusplus.h
```

**Review by knst — APPROVED** [review]
urACK

**Comment by UdjinM6** [comment]
rebased from GH GUI to fix `Merge Fast-Forward Only` check

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit
